### DISCUSSION TITLE
updateTrigger with Vue.js app

### DISCUSSION THREAD

**Discussion by botivegh** [visgl/deck.gl#6523] [Q&A] [answered]
Hey,

I've just started experimenting with Vue.js + deck.gl + Carto according the this sample project:
[https://github.com/CartoDB/viz-doc/tree/master/deck.gl/examples/pure-js/vue/carto3](https://github.com/CartoDB/viz-doc/tree/master/deck.gl/examples/pure-js/vue/carto3)

I'm having trouble making the `updateTrigger` property work on the` getFillColor` property of a `GeojsonLayer` object. The `changeColor()` method is successfully changing the `highlightColor` prop but the `getFillColor` prop change has no effect, even though I use the `updateTrigger` property. I feel like I have a conceptual misunderstanding on deck.gl or Vue.js and cannot figure out what is the issue. Any help would be much appreciated! 

For reference/more context please ask or visit the example project mentioned above.

**Home.vue**

```
 <template>
  <div class="home-container">
    <button @click="changeColor()">TEST</button>
    <ViewTemplate></ViewTemplate>
  </div>
</template>

<script>
import { GeoJsonLayer } from "@deck.gl/layers";
import ViewTemplate from "@/components/view-template/ViewTemplate.vue";
import layerManager from "@/components/map-component/map-utils/layerManager";

export default {
  name: "app-home",
  components: {
    ViewTemplate,
  },
  data: () => ({
    storesData: [],
    color1: [255, 255, 255],
    color2: [255, 255, 0],
    highlight: [0, 255, 0],
  }),
  mounted() {
    setDefaultCredentials(this.credentials);

    const storesQuery = "SELECT * FROM retail_stores";
    const storesUrl = `https://public.carto.com/api/v2/sql?q=${storesQuery}&format=geojson`;
    layerManager.addLayer(
      new GeoJsonLayer({
         id: "stores",
        data: storesUrl,
        pointRadiusUnits: "pixels",
        lineWidthUnits: "pixels",
        pickable: true,
        autoHighlight: true,
        highlightColor: this.highlight,
        getFillColor: (object) =>
          object.properties.size_m2 > 500 ? this.color1 : this.color2,
        updateTriggers: {
          getFillColor: [this.color1, this.color2],
        },
        onDataLoad: (data) => (this.storesData = data.features),
      })
    );
  methods: {
    changeColor() {
      this.color1 = [0, 0, 255];
      this.color2 = [0, 255, 255];
      this.highlight = [255, 0, 0];

      layerManager["updateLayer"]("stores", {
        getFillColor: (object) =>
          object.properties.size_m2 > 500 ? this.color1 : this.color2});
      layerManager["updateLayer"]("stores", { highlightColor: this.highlight });
    },
  },
</script>
```

[**layerManager.js** ](https://github.com/CartoDB/viz-doc/blob/master/deck.gl/examples/pure-js/vue/carto3/src/components/map-component/map-utils/layerManager.js)
```
import DeckMap from './DeckMap'
export default {

  updateDeckInstance() {
    if (!this.deckInstance) {
      return
    }
    console.log('updateDeckInstance')
    const layers = [...Object.values(this.layers)];
    if (this.deckInstance) {
      this.deckInstance.setProps({ layers })
    }
  },
  updateLayer(id, props) {
    if (!this.layers[id]) {
      return
    }
    const layer = this.layers[id]
    this.layers[id] = layer.clone(props)
    this.updateDeckInstance()
  },
}
```

![image](https://user-images.githubusercontent.com/47627720/146782642-f3f81ced-1305-4c4b-80ff-30299699b923.png)

**Comment by Pessimistress** [comment, marked as answer]
Your `updateTriggers` is only set once to the color values on initial mount. When you want to reevaluate the fill colors you should update the `updateTriggers` prop instead of the `getFillColor` prop. See [example](https://deck.gl/docs/developer-guide/performance#use-updatetriggers).

**Reply by botivegh** [reply]
Thank you! That makes complete sense now.
